### PR TITLE
🎉 Source Stripe: Add `PaymentIntents` stream

### LINKED ISSUE #5568: Stripe Source - Retrieve PaymentIntent source
## Tell us about the problem you're trying to solve
Would like to pull PaymentIntents from Stripe. https://stripe.com/docs/api/payment_intents

## Describe the solution you’d like
for the existing Stripe connector to be able to pull PaymentIntents. 

## Describe the alternative you’ve considered or used
no alternative at the moment

## Additional context
-

## Are you willing to submit a PR?
can be discussed :)

### PR DESCRIPTION
**Pull request by Zirochkaa**
## What
Closes #5568.
I removed `/source-stripe/integration_tests/expected_subscriptions_records.txt` file because data for `subscriptions` and `customers` streams keep on changing, especially following fields:
- current_period_end;
- current_period_start;
- latest_invoice;

## Pre-merge Checklist
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

**Review by avida — APPROVED** [review]

File: airbyte-integrations/connectors/source-stripe/setup.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 from setuptools import find_packages, setup
 
-MAIN_REQUIREMENTS = ["airbyte-cdk", "stripe"]
+MAIN_REQUIREMENTS = ["airbyte-cdk~=0.1", "stripe"]
```
Comment (htrueman) [inline airbyte-integrations/connectors/source-stripe/setup.py:28]: could you specify the version of `stripe` lib too?

Comment (htrueman) [inline airbyte-integrations/connectors/source-stripe/setup.py:28]: Could you specify the version of stripe lib too?

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/payment_intents.json [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    },
+    "amount_capturable": {
+      "type": ["null", "integer"]
+    },
+    "amount_received": {
+      "type": ["null", "integer"]
+    },
+    "application": {
+      "type": ["null", "string", "object"],
+      "properties": {}
```
Comment (htrueman) [inline airbyte-integrations/connectors/source-stripe/source_stripe/schemas/payment_intents.json:21]: Why all `properties` objects are blank there?

File: airbyte-integrations/connectors/source-stripe/source_stripe/source.py [hunk 3]
```diff
@@ -101,7 +101,8 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
     def request_params(self, stream_state=None, **kwargs):
```
Comment (htrueman) [inline airbyte-integrations/connectors/source-stripe/source_stripe/source.py:101]: type annotation is missed

File: airbyte-integrations/connectors/source-stripe/source_stripe/source.py [hunk 4]
```diff
@@ -101,7 +101,8 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
...
         stream_state = stream_state or {}
         params = super().request_params(stream_state=stream_state, **kwargs)
-        params["created[gte]"] = stream_state.get(self.cursor_field)
+        if stream_state:
```
Comment (htrueman) [inline airbyte-integrations/connectors/source-stripe/source_stripe/source.py:104]: perhaps you should also check if `cursor_field` exist?
```suggestion
        if stream_state and stream_state.get(self.cursor_field):
```

File: airbyte-integrations/connectors/source-stripe/source_stripe/source.py (hunk 3 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-stripe/source_stripe/source.py:101]: Done.

File: airbyte-integrations/connectors/source-stripe/source_stripe/source.py (hunk 4 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-stripe/source_stripe/source.py:104]: Done.

File: docs/integrations/sources/stripe.md [hunk 5]
```diff
@@ -26,6 +26,7 @@ This Source is capable of syncing the following core Streams:
 * [Subscriptions](https://stripe.com/docs/api/subscriptions/list) \(Incremental\)
 * [Subscription Items](https://stripe.com/docs/api/subscription_items/list)
 * [Transfers](https://stripe.com/docs/api/transfers/list) \(Incremental\)
```
Comment (sherifnada) [inline docs/integrations/sources/stripe.md:28]: could you insert this in alphabetical order?

**Review by sherifnada — APPROVED** [review]

File: docs/integrations/sources/stripe.md (hunk 5 above)
Comment (Zirochkaa) [reply docs/integrations/sources/stripe.md:28]: Sure, done.

File: airbyte-integrations/connectors/source-stripe/setup.py (hunk 1 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-stripe/setup.py:28]: Done.

File: airbyte-integrations/connectors/source-stripe/source_stripe/schemas/payment_intents.json (hunk 2 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-stripe/source_stripe/schemas/payment_intents.json:21]: Some of them don't include descriptions in the documentation.

**Pull request merged**